twitfix.py

```diff
@@ -52,18 +53,22 @@ def oembedend():
 
 @app.route('/<path:subpath>')
 def twitfix(subpath):
+    user_agent = request.headers.get('user-agent')
+    if user_agent in discord_user_agents:
 
-    match = pathregex.search(subpath)
-    if match is not None:
-        twitter_url = subpath
+        match = pathregex.search(subpath)
+        if match is not None:
+            twitter_url = subpath
 
-        if match.start() == 0:
-            twitter_url = "https://twitter.com/" + subpath
+            if match.start() == 0:
+                twitter_url = "https://twitter.com/" + subpath
 
-        res = embedVideo(twitter_url)
-        return res
+            res = embedVideo(twitter_url)
+            return res
+        else:
+            return render_template('default.html', message="This doesn't seem to be a twitter link, try /other/ to see if other kinds of video link works? (experimental)")
     else:
-        return render_template('default.html', message="This doesn't seem to be a twitter link, try /other/ to see if other kinds of video link works? (experimental)")
+        return redirect("https://twitter.com/" + subpath, 301)
```

Comment:
Subpath is not garunteed to be only the last half of a url, as you can use [host url]/[full twitter url], using it this way causes the redirect to point to a misformatted url (
![image](https://user-images.githubusercontent.com/12601774/125377139-0bd0f880-e352-11eb-9b06-77661c3f29ea.png)
)
Change this line to ``return redirect( twitter_url, 301 )` instead to get the preformatted url
Other than that, I tried it out and it works a lot better!

Reply:
I'll make that fix in the morning, ty!